src/analyzing/AST_to_IL.ml

```diff
@@ -688,6 +704,38 @@ and expr_aux env ?(void = false) e_gen =
         | _ -> exp
       in
       ident_function_call_hack exp
+  (* x = ClassName(args ...) in Python *)
+  (* ClassName has been resolved to __init__ by the pro engine. *)
+  (* Identified and treated as x = New ClassName(args ...) to support
+     field sensitivity. See HACK(new) *)
+  | G.Assign
+      ( ({
+           e =
+             G.N
+               (G.Id ((_, _), { id_type = { contents = Some ret_ty }; _ }) as
+                obj);
+           _;
+         } as obj_e),
+        _,
+        ({
+           e =
+             G.Call
+               ( {
+                   e =
+                     ( G.N (Id (_, id_info))
+                     | G.DotAccess (_, _, FN (Id (_, id_info))) );
```

Comment:
Can this be a `DotAccess` ? In which case ? A qualified name ? Good to clarify in a comment.